test/conftest.py

```diff
@@ -61,6 +62,45 @@ def pytest_addoption(parser):
     )
 
 
+@pytest.hookimpl(tryfirst=True)
+def pytest_runtest_makereport(item, call):
+    """Make timeouts appear as such in the logs.
+
+    The pytest-timeout plug-in messes with the durations that are reported for tests.
+    Fix that by marking the report such that we can print `TIMEOUT` instead of
+    `FAILED (incorrect duration)`. See
+    https://github.com/pytest-dev/pytest/blob/38d8deb74d95077ebf189440ca047e14f8197da1/src/_pytest/runner.py#L366
+    """
+    timed_out = False
+    try:
+        if call.excinfo is not None:
+            msg = call.excinfo.value.msg
+            if isinstance(msg, str) and msg.startswith('Timeout >'):
+                timed_out = True
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I'm not sure i understand what this block is doing. Please could you explain a little bit? Also, could exceptions do you see that get risen here? Perhaps they should be coded in to avoid the broad-except.

Reply:
Have a look at the default implementation (url): it is a one-liner that creates and returns the report. We introduce additional code here, so we definitely do not want weird exceptions cause more problems. Exceptions should not occur, and when the test has timed out (or the test failed, which is actually a caught `AssertionError` most of the time) everything proceeds as pytest does it normally. The only purpose of this block is to add a magic attribute to the report so that we can later tell that it was a timeout. The exception info that is part of the `call` is dropped (see the constructor/class method) when creating the report, I think. So our only chance is to catch that here and store our findings. Then we can indeed use that later on in the `pytest_report_teststatus` function.

Reply:
I know that this is a broad except and right now we are not even checking our tests for stuff like that, but we really want this here. Hence I already added the `pylint: disable=` comment :)

Reply:
What exceptions do you expect would be risen here?

Reply:
Could be `AttributeError`s if `call.excinfo` is not of the type it should be according to `pytest`. Or if `call.excinfo.value` is not an Exception and hence doesn't have a `msg` attribute. So there shouldn't be any errors here. I still don't want to risk removing the `try: ... except`.

Reply:
Let's not use `except Exception` here. Please make it `AttributeError` if that's all you expect.

Reply:
That piece of code is outdated. The latest version of the commit doesn't even have a `try` in it...

Reply:
Ah ok :)